api/apps/api/test/access-control/project-acl.fixtures.ts

```diff
@@ -0,0 +1,41 @@
+import * as request from 'supertest';
+import { bootstrapApplication } from '../utils/api-application';
+import { GivenUserIsLoggedIn } from '../steps/given-user-is-logged-in';
+import { GivenProjectExists } from '../steps/given-project';
+import { Repository } from 'typeorm';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { PublishedProject } from '@marxan-api/modules/published-project/entities/published-project.api.entity';
+
+export const getFixtures = async () => {
+  const app = await bootstrapApplication();
+  const randomUserToken = await GivenUserIsLoggedIn(app);
+  const publishedProjectsRepo: Repository<PublishedProject> = app.get(
+    getRepositoryToken(PublishedProject),
+  );
+  const cleanups: (() => Promise<void>)[] = [];
+
+  return {
+    cleanup: async () => {
+      await Promise.all(cleanups.map((clean) => clean()));
```

Comment:
most likely you would want to `reverse` cleanup (like, creating project->scenario->roles would have intention to be "cleaned" in reverse way, starting with roles, not project)

Reply:
What do you mean here? Doesn't `cleanup` just clean everything up in the DB? Or am I understanding this wrong?

I didn't make any changes in here as it looks like it was working properly (deleting what I need for every test on every cycle). But if you can explain it a little bit I would appreciate it very much 🙇

Reply:
@rubvalave I like the sandwich process creation as an example:

bread > butter > cheese > ham > tomato

now if you will try to remove bread first (as in the code), everything else may fall on the floor - thats why cleanup `reverse` (in terms of order) is more safe to execute.

Reply:
@rubvalave what Kamil suggests can be as simple as in the snippet below (lifted wholesale from `input-file.e2e-spec.ts`). What you have here may work now because there may be no dependencies, but if when adding new tests/setups dependencies are introduced, a LIFO approach should be safer.

```
      for (const cleanup of cleanups.reverse()) {
        await cleanup();
      }
```